Taglierino.hs

```diff
@@ -910,12 +945,95 @@ honestRange agents =
   $ [pretty (agentLabel a) <> dot <> brackets (pretty $ "0.." ++ show (i-1))
     |(a, i) <- agents]
 
-
+makeStore :: Set Term -> Set Term -> String -> Int -> LTS.Process
+makeStore allowed records agentName sessions= 
+  LTS.Process ("Store_" ++ agentName) Nothing False initialState body alphabet
+  where 
+    combinations :: Int -> [a] -> [[a]]
+    combinations 0 _ = [[]]
+    combinations n xs = [y : ys | y:xs' <- tails xs, ys <- combinations (n - 1) xs']
+    
+    allComb :: Ord a => [a] -> [S.Set a]
+    allComb l = map S.fromList $ concat result
+      where len = length l
+            result = map (`combinations`  l) [0.. len]
+
+    entries :: [S.Set Term]
+    entries = allComb $ S.toList records
```

Comment:
I think `Data.Set.powerSet` does exactly this.

Reply:
I add powerset but commented it out to keep the generated model the same.